api/file.go

```diff
@@ -0,0 +1,193 @@
+package api
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/konveyor/tackle2-hub/auth"
+	"github.com/konveyor/tackle2-hub/migration/v3/model"
+	"io"
+	"mime"
+	"net/http"
+	"os"
+	pathlib "path"
+	"time"
+)
+
+//
+// Routes
+const (
+	FilesRoot = "/files"
+	FileRoot  = FilesRoot + "/:" + ID
+)
+
+//
+// FileHandler handles file routes.
+type FileHandler struct {
+	BaseHandler
+}
+
+//
+// AddRoutes adds routes.
+func (h FileHandler) AddRoutes(e *gin.Engine) {
+	routeGroup := e.Group("/")
+	routeGroup.Use(auth.Required("files"))
+	routeGroup.GET(FilesRoot, h.List)
+	routeGroup.GET(FilesRoot+"/", h.List)
+	routeGroup.POST(FileRoot, h.Create)
+	routeGroup.PUT(FileRoot, h.Create)
+	routeGroup.GET(FileRoot, h.Get)
+	routeGroup.DELETE(FileRoot, h.Delete)
+}
+
+// List godoc
+// @summary List all files.
+// @description List all files.
+// @tags get
+// @produce json
+// @success 200 {object} []api.FileField
```

Comment:
The doc strings all refer to `api.FileField` where they mean to refer to the `api.File` resource type.